### ISSUE TITLE
Home/End are not mapped with Ctrl+A/E on iTerm when "Use PC Style Home/End #2" is checked

### ISSUE THREAD

**Issue by astorije** [tekezo/Karabiner#495]
Although Karabiner says:

> (+ Change Home to Control+A in Emacs,Terminal,X11)
> (+ Change End to Control+E in Emacs,Terminal,X11)

and, according to https://github.com/tekezo/Karabiner/issues/460, iTerm is already in `TERMINAL`, when this option is checked and Home/End are pressed on an external keyboard, they send <kbd>⌘</kbd>+<kbd>←</kbd> and <kbd>⌘</kbd>+<kbd>→</kbd> instead of <kbd>Ctrl</kbd>+<kbd>A</kbd> and <kbd>Ctrl</kbd>+<kbd>E</kbd>.

I can confirm the <kbd>Home</kbd> and <kbd>End</kbd> keys do map correctly to <kbd>Ctrl</kbd>+<kbd>A</kbd> and <kbd>Ctrl</kbd>+<kbd>E</kbd> when this option is disabled, so it doesn't come from the keyboard, this time...

Any idea what's going on?

**Comment by tekezo** [comment]
I cannot reproduce this issue.
Please give me more information.

1. Please show me your enabled settings in Karabiner. (Screen shot of Karabiner Preferences with "show enabled only".)
2. Please confirm App tab in EventViewer. https://pqrs.org/osx/karabiner/document.html.en#eventviewer

The current appnames should be "TERMINAL" when the current application is iTerm.
Is it same on your machine?
![151021-0001](https://cloud.githubusercontent.com/assets/659178/10630159/d11b5b4e-780f-11e5-84b2-54483e72f169.png)

**Comment by eduncan911** [comment]
I've got the same issue.

Karabiner 10.15.0

Karabiner reports "TERMINAL" in the event viewer.  But, the Home/End keys absolutely changes in iTerm2 from selecting the "Use PC Style Home/End #2" selection.  So, it continues to affect iTerm 2.

My iTerm2 is basically default settings, latest RTM.

I also switched to iTerm 2 Beta 2.9.20160206, and it acts exactly the same way.

Finally, i deleted my iTerm 2 settings file - restoring to defaults - and it still continues to switch when Karabiner has Home/End #2 selected.

![screen shot 2016-02-26 at 1 25 24 am](https://cloud.githubusercontent.com/assets/104980/13344853/dacac25c-dc27-11e5-9be5-e821fa877a25.png)

**Comment by tekezo** [comment]
@eduncan911 `Use PC Style Home/End #2` changes home and end keys to control-a and control-e in terminal.
Do you want to pass raw home and end key event to terminal?

**Comment by eduncan911** [comment]
I'm new to devving on OS X directly using terminals (am a long time Linux guru running in VMs for years, trying to switch to native os x).  For me, I use CTRL-A often for Screen on local and in SSH sessions.

Wouldn't that conflict with how I use Control-A?

Also, VIM doesn't seem to like CTRL-A and CTRL-E.. it modifies the buffer when all I am trying to do is get to the end of the line.

In this option, it says `Except in Virtual Machine, RDC, VNC, TeamViewer, EMACS, `

![screen shot 2016-02-26 at 9 32 28 am](https://cloud.githubusercontent.com/assets/104980/13354925/e0a22730-dc6b-11e5-81cb-a3078904fc75.png)

Doesn't that mean the Home/End modifications should be ignored by TERMINAL?

When I de-select this option, iTerm 2 allows me to use Home/End on my PC-style keyboard like normal.  Given though, OS X has serious problems with Home/End keys.  Hence, why i am trying to enable this feature for anything-else-except-Terminal.

**Comment by eduncan911** [comment]
Ok, I think I am seeing the problem now...  Now I understand the screenshot above where it says:

```
(+ Change Home to Control-A in Emacs,Terminal,X11)
(+ Change End to Control-E in Emacs,Terminal,X11)
```

I do not want this.  But, I do desire the `Command+Left/Right` functionality of Home/End in OS X applications, as this is how Linux and Windows normally uses Home/End.

How can I make `Use PC Style Home/End #2` ignore/not send CTRL-A/E for Terminal?

Optionally, how to make this #2 option send Home/End key's as CMD+Left/Right?  

^- I think this is the best option, as I have a macbook pro and switch from PC keyboard to portable laptop quite often. 

Basically, it comes down to that I want the functionality that OSX Command+Left/Right provides in OSX by using the PC-style keyboard's Home/End keys. 

I can setup iTerm to use CMD+left/right as Home/End escape codes, which works for me.

**Comment by tekezo** [comment]
@eduncan911 
Overwriting home and end key behavior is the easiest way.

Please use new setting by this private.xml together.
https://github.com/tekezo/Files/blob/master/Karabiner/private.terminal_pass_through_home_end/private.xml

How to use private.xml:
https://pqrs.org/osx/karabiner/document.html#privatexml

You can change home and end key to command+left and right by uncomment these lines.
https://github.com/tekezo/Files/blob/master/Karabiner/private.terminal_pass_through_home_end/private.xml#L10-L13

**Comment by eduncan911** [comment]
Perfect, that works with the `Use PC Style Home/End # 2` setting in iTerm 2 and Vim!

Though, uncommenting those lines didn't seem to do much.  I left them commented out for now.

Though I wasn't the one who opened this issue, I'd say this resolves the issue and it can be closed.  :+1:
